cJSON.h

```diff
@@ -81,85 +81,122 @@ typedef struct cJSON_Hooks
       void (*free_fn)(void *ptr);
 } cJSON_Hooks;
 
+#if !defined(__WINDOWS__) && (defined(WIN32) || defined(WIN64) || defined(_MSC_VER) || defined(_WIN32))
+#define __WINDOWS__
+#endif
+#ifdef __WINDOWS__
+
+/* When compiling for windows, we specify a specific calling convention to avoid issues where we are being called from a project with a different default calling convention.  For windows you have 2 define options:
+
+CJSON_DECLARE_STATIC - Define this in the case where you don't want to ever dllexport symbols
```

Comment:
I'm not sure if I understand correctly. But this is not related to building static or dynamic libraries, right?

In this case I think `CJSON_HIDE_SYMBOLS` would be a better name and it could also set the visibility to hidden for the *nix compilers. Or did I misunderstand that?